core/src/main/scala/cromwell/util/JsonEditor.scala

```diff
@@ -2,98 +2,67 @@ package cromwell.util
 
 import cats.data.NonEmptyList
 import common.collections.EnhancedCollections._
-import io.circe.ACursor
-import io.circe.Json
+import io.circe.{Json, JsonNumber, JsonObject}
 import mouse.all._
-import cats.syntax.traverse._
-import cats.syntax.functor._
-import cats.instances.list._
-import cats.data.State
+import io.circe.Json.Folder
+
+import scala.collection.immutable
 
 object JsonEditor {
 
-  def includeExcludeJson(json: Json, includeKeys: Option[NonEmptyList[String]], excludeKeys: Option[NonEmptyList[String]]): Json = {
+  def includeExcludeJson(json: Json, includeKeys: Option[NonEmptyList[String]], excludeKeys: Option[NonEmptyList[String]]): Json =
     (includeKeys, excludeKeys) match {
       // Take includes, and then remove excludes
       case (Some(includeKeys), Some(excludeKeys)) => includeJson(json, includeKeys) |> (excludeJson(_, excludeKeys))
       case (None, Some(excludeKeys)) => excludeJson(json, excludeKeys)
       case (Some(includeKeys), None) => includeJson(json, includeKeys)
       case _ => json
     }
-  }
 
-  private def excludeKeys(keys: NonEmptyList[String]) : State[ACursor, Unit] = State.modify{cursor =>
-    val levelKeys: List[String] = cursor.keys.toList.flatMap(_.toList)
-
-    // Take each key in the json level and turn it into a state operation over the cursor
-    val modifications: State[ACursor, Unit] = levelKeys.traverse{key =>
-      val delete = keys.foldLeft(false)(_ || key.startsWith(_))
-      State.modify[ACursor] { cursor =>
-        if (delete)
-          //moves cursor back to parent
-          cursor.downField(key).delete
-        else {
-          val newCursor = cursor.downField(key)
-          val eval = excludeKeys(keys).run(newCursor)
-          //ignoring unit output
-          val (output,_) = eval.value
-          //we're in the field, have to go back to the parent for the next field to evaluate
-          output.up
-        }
+  def includeJson(json: Json, keys: NonEmptyList[String]): Json = {
+    def folder: Folder[(Json, Boolean)] = new Folder[(Json, Boolean)] {
+      override def onNull: (Json, Boolean) = (Json.Null, false)
+      override def onBoolean(value: Boolean): (Json, Boolean) = (Json.fromBoolean(value), false)
+      override def onNumber(value: JsonNumber): (Json, Boolean) = (Json.fromJsonNumber(value), false)
+      override def onString(value: String): (Json, Boolean) = (Json.fromString(value), false)
+      override def onArray(value: Vector[Json]): (Json, Boolean) = {
+        val newArrayAndKeeps: immutable.Seq[(Json, Boolean)] = value.map(_.foldWith(folder))
+        val keep: Boolean = newArrayAndKeeps.map(_._2).foldLeft(false)(_ || _)
+        (Json.fromValues(newArrayAndKeeps.map(_._1)), keep)
```

Comment:
ToL-only: Generally I'm not a fan of the tuple `._2` syntax, preferring either custom classes or the verbose `case (_, descriptive) => descriptive`, but this seems to work.

Reply:
yeah me neither :).  I"ll fix it